inference/core/workflows/core_steps/formatters/vlm_as_classifier/v1.py

```diff
@@ -0,0 +1,247 @@
+import json
+import logging
+import re
+from typing import Dict, List, Literal, Optional, Tuple, Type, Union
+from uuid import uuid4
+
+from pydantic import ConfigDict, Field
+
+from inference.core.workflows.execution_engine.entities.base import (
+    OutputDefinition,
+    WorkflowImageData,
+)
+from inference.core.workflows.execution_engine.entities.types import (
+    BATCH_OF_BOOLEAN_KIND,
+    BATCH_OF_CLASSIFICATION_PREDICTION_KIND,
+    BATCH_OF_STRING_KIND,
+    LIST_OF_VALUES_KIND,
+    STRING_KIND,
+    ImageInputField,
+    StepOutputImageSelector,
+    StepOutputSelector,
+    WorkflowImageSelector,
+    WorkflowParameterSelector,
+)
+from inference.core.workflows.prototypes.block import (
+    BlockResult,
+    WorkflowBlock,
+    WorkflowBlockManifest,
+)
+
+JSON_MARKDOWN_BLOCK_PATTERN = re.compile(
+    r"```json\n([\s\S]*?)\n```", flags=re.IGNORECASE
+)
+
+LONG_DESCRIPTION = """
+The block expects string input that would be produced by blocks exposing Large Language Models (LLMs) and 
+Visual Language Models (VLMs). Input is parsed to classification prediction which is output from the 
+block.
+
+Accepted formats:
+- valid JSON strings
+- JSON documents wrapped with Markdown tags (very common for GPT responses)
+```
+```json
+{"my": "json"}
+```
+```
+
+**Details regarding block behaviour:**
+
+- `error_status` is set `True` whenever parsing cannot be completed
+
+- in case of multiple markdown blocks with raw JSON content - only first will be parsed
+"""
+
+SHORT_DESCRIPTION = "Parses raw string into classification prediction."
+
+
+class BlockManifest(WorkflowBlockManifest):
+    model_config = ConfigDict(
+        json_schema_extra={
+            "name": "VLM as Classifier",
+            "version": "v1",
+            "short_description": SHORT_DESCRIPTION,
+            "long_description": LONG_DESCRIPTION,
+            "license": "Apache-2.0",
+            "block_type": "formatter",
+        }
+    )
+    type: Literal["roboflow_core/vlm_as_classifier@v1"]
+    image: Union[WorkflowImageSelector, StepOutputImageSelector] = Field(
+        description="The image which was the base to generate VLM prediction",
+        examples=["$inputs.image", "$steps.cropping.crops"],
+    )
+    vlm_output: StepOutputSelector(kind=[BATCH_OF_STRING_KIND]) = Field(
+        description="The string with raw classification prediction to parse.",
+        examples=[["$steps.lmm.output"]],
+    )
+    classes: Union[
+        WorkflowParameterSelector(kind=[LIST_OF_VALUES_KIND]),
+        StepOutputSelector(kind=[LIST_OF_VALUES_KIND]),
+        List[str],
+    ] = Field(
+        description="List of all classes used by the model, required to "
+        "generate mapping between class name and class id.",
+        examples=[["$steps.lmm.classes", "$inputs.classes", ["class_a", "class_b"]]],
+    )
```

Comment:
Open question:
* shall we specify dropdown to point model type and task type - I expect that in the future we may need to use that values to choose parsing method specific for given (model, task) pair, yet now it is not needed and would enforce people to choose the option.

Reply:
can the model type not be inferred from the previous block?

Reply:
not really if we want to fail-fast in compilation, given that there is wrong setup for model/task which is not supported. (compiler do not evaluate steps outputs - hence we would have selector to output not value)

I take the hustle of setting it manually as worth taking if you save on execution costs of API calls only to discover that your workflow is invalid

Reply:
Hmm, I am not sure how worfklows works, but can't we have an output type for each llm string, and then make sure that output type is compatible with this block?

Reply:
Like the way the blocks are hooked up should be known at compile time so this model_type should be known at compile time, right?

Reply:
* output type (kind) for each would solution, but would also be an overkill
* compiler do not understand types of blocks, but kinds of data